### PR TITLE
W-9004683-Changes to use the common client creds with username and pa…

### PR DESCRIPTION
**Pull request by palaniappa**
…ssword flow

### CODE REVIEW

File: src/main/java/com/salesforce/cdp/queryservice/util/TokenHelper.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 import com.google.common.cache.Cache;
 import com.google.common.cache.CacheBuilder;
+import com.google.gson.Gson;
```
Comment (g1-pai) [inline src/main/java/com/salesforce/cdp/queryservice/util/TokenHelper.java:21]: nit - seems like unused import

File: src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java [hunk 2]
```diff
@@ -38,9 +38,34 @@ public QueryServiceConnection(String url, Properties properties) {
         this.properties = properties;
         this.serviceRootUrl = url.substring(Constants.DATASOURCE_TYPE.length());
         this.properties.put(Constants.LOGIN_URL, serviceRootUrl);
+        setupDefaultClientSecretsIfRequired(serviceRootUrl, this.properties);
         this.queryExecutor = new QueryExecutor(properties);
     }
 
+    private void setupDefaultClientSecretsIfRequired(String serviceRootUrl, Properties properties) {
+        if(properties.containsKey(Constants.USER) && !properties.containsKey(Constants.USER_NAME)) {
+            properties.put(Constants.USER_NAME, properties.get(Constants.USER));
```
Comment (g1-pai) [inline src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java:47]: Do we need new User constant? Can we stick only to userName standard?

File: src/main/java/com/salesforce/cdp/queryservice/util/Constants.java [hunk 3]
```diff
@@ -33,6 +33,19 @@
     public static final String DRIVER_NAME = "QueryService-jdbc";
     public static final String DRIVER_VERSION = "1.0";
 
+    // Common client id and secret information
+    public static final String STMPA_SERVER_URL = "stmpa.stm.salesforce.com";
+    public static final String STMPA_DEFAULT_CLIENT_ID = "3MVG9Iu66FKeHhIMTw4_fbTKdbfgYVQXSsSJ6jOMZWGwgJ0RVhaGO3_RXBTxmWUHholNWtSgpa6nmjNeOvdBX";
+    public static final String STMPA_DEFAULT_CLIENT_SECRET = "9662F811A0702D3FE62BCF2AF3A51D06F38E1D0FA1573E5D17301C9FA4B216BB";
+
+    public static final String STMPB_SERVER_URL = "stmpb.stm.salesforce.com";
+    public static final String STMPB_DEFAULT_CLIENT_ID = "3MVG9VeAQy5y3BQVJqaUbFmV5jd8imcck2K5idmrTTGocSu9qZZ6qkbuEkxECKVYwmzm3WgvxkujqsxZDcBpL";
```
Comment (g1-pai) [inline src/main/java/com/salesforce/cdp/queryservice/util/Constants.java:42]: Client Id of STMPB is 3MVG9Iu66FKeHhIPHKRhbohYVNiUUHCt2h_HeUwYwzEdrn4unY9uMKKdfYeGnb19ngLq6T6Q8hwEHVexROWav

File: src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java [hunk 4]
```diff
@@ -38,9 +38,34 @@ public QueryServiceConnection(String url, Properties properties) {
...
+        }
+
+        if(properties.containsKey(Constants.USER_NAME)
+                && !properties.containsKey(Constants.CLIENT_ID)
```
Comment (g1-pai) [inline src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java:51]: Can we add this in tokenHelper as these are related to tokens?

File: src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java (hunk 2 above)
Comment (palaniappa) [reply src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java:47]: with DBeaver by filling in just the user name and password we should be able to connect. So the default property key for username is user.

File: src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java (hunk 4 above)
Comment (palaniappa) [reply src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java:51]: Felt it is more appropriate in connection object that gives token helper the way/format it needs.

File: src/main/java/com/salesforce/cdp/queryservice/util/Constants.java (hunk 3 above)
Comment (palaniappa) [reply src/main/java/com/salesforce/cdp/queryservice/util/Constants.java:42]: fixed. thanks for catching this.

**Review by g1-pai — APPROVED** [review]

**Pull request merged**